extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -104,20 +106,35 @@ export class AmpUserNotification extends AMP.BaseElement {
     this.elementId_ = assert(this.element.id,
         'amp-user-notification should have an id.');
 
-    assert(this.element.hasAttribute('data-show-if-href'),
-        `"amp-user-notification" (${this.elementId_}) ` +
-        'should have "data-show-if-href" attribute.');
-    /** @private @const {string} */
-    this.showIfHref_ = assertHttpsUrl(
-        this.element.getAttribute('data-show-if-href'), this.element);
+    /** @const @private {!Promise<!Storage>} */
+    this.storagePromise_ = storageFor(this.win_);
 
-    assert(this.element.hasAttribute('data-dismiss-href'),
-        `"amp-user-notification" (${this.elementId_}) ` +
-        'should have "data-dismiss-href" attribute.');
+    /** @private @const {?string} */
+    this.storageKey_ = isExperimentOn(this.getWin(), 'amp-storage') ?
+        'amp-user-notification:' + this.elementId_ : null;
 
-    /** @private @const {string} */
-    this.dismissHref_ = assertHttpsUrl(
-        this.element.getAttribute('data-dismiss-href'), this.element);
+    /** @private @const {?string} */
+    this.showIfHref_ = this.element.getAttribute('data-show-if-href');
```

Comment:
can we either add a note here or on the PR to update validator. (or create an issue)